src/src/streamapi/hw/hillel/Main.java

```diff
@@ -27,10 +28,17 @@ public static List<Product> getExpensiveBooks(List<Product> products) {
                 .collect(Collectors.toList());
     }
 
+    public static List<Product> getDiscountedBooks(List<Product> products) {
+        return products.stream()
+                .filter(product -> product.getType().equals("Book") && product.hasDiscount())
+                .peek(product -> product.setPrice(product.getPrice() * 0.9))
```

Comment:
replace `peek` with `map` as `peek` is used for testing/debugging purposes only. If your code won't compile after the change, it is totally ok, please find a solution.